### PR TITLE
add option to `oc whoami` that prints server url

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://github.com/openshift/origin/issues/11148

Currently there is no "simple" way to obtain the rest api's URL.
This patch adds an option `--show-server` || `-u` to `oc whoami`
that retrieves the server url from the first cluster in `config.Cluster`

**Before**

```
$ oc config view --minify -o jsonpath='{.clusters[*].cluster.server}'
https://10.13.137.149:8443% # no linebreak at the end
```

**After**

```
$ oc whoami -u # or --show-server
https://10.13.137.149:8443
```

cc @openshift/cli-review @smarterclayton

### CODE REVIEW

**Comment by liggitt** [comment]
--show-server to parallel the other options?

**Comment by liggitt** [comment]
Not sure a short flag is needed

**Comment by juanvallejo** [comment]
@liggitt Thanks for the feedback, comments addressed https://github.com/openshift/origin/pull/11180/commits/85e2d6b74dc53bb88c6680beab513ecf2d20e612

File: pkg/cmd/cli/cmd/whoami.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func NewCmdWhoAmI(name, fullName string, f *clientcmd.Factory, out io.Writer) *c
 
 	cmd.Flags().BoolP("show-token", "t", false, "Print the token the current session is using. This will return an error if you are using a different form of authentication.")
 	cmd.Flags().BoolP("show-context", "c", false, "Print the current user context name")
+	cmd.Flags().Bool("show-server", false, "Print the server's REST API URL")
```
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/whoami.go]: "Print the current server's REST API URL"

File: pkg/cmd/cli/cmd/whoami.go [hunk 2]
```diff
@@ -89,6 +90,17 @@ func RunWhoAmI(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []s
...
+	if kcmdutil.GetFlagBool(cmd, "show-server") {
+		cfg, err := f.OpenShiftClientConfig.RawConfig()
+		if err != nil {
+			return err
+		}
+		for _, c := range cfg.Clusters {
+			fmt.Fprintf(out, "%s\n", c.Server)
+			return nil
+		}
+		return fmt.Errorf("unable to get clusters. Cannot retrieve server url")
```
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/whoami.go]: "unable to get clusters. Cannot retrieve server URL."

**Comment by juanvallejo** [comment]
@fabianofranz thanks for the feedback! Review comments addressed, went ahead and squashed

**Comment by fabianofranz** [comment]
LGTM [merge]

**Comment by juanvallejo** [comment]
merge tests flaked on AWS failure: https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/6660/consoleFull#-27859736956a6674fe4b089752bb4314c

**Pull request merged**